src/comparePayloads.js

```diff
@@ -0,0 +1,180 @@
+export function comparePayloads(payload, comparedPayload) {
+  if (payload.id === comparedPayload.id) {
+    throw new Error(
+      `You try to compare two payloads with the same id '${payload.id}'`
+    );
+  }
+
+  if (payload.rootDependencyName !== comparedPayload.rootDependencyName) {
+    throw new Error(
+      `You can't compare different package payloads '${payload.rootDependencyName}' and '${comparedPayload.rootDependencyName}'`
+    );
+  }
```

Comment:
I wonder if we should also skip (returns 0 diff, not throw) if the two payload are the same root dependency version ?
Something like this:
```js
function payloadRootVersions(payload) {
  return Object.keys(payload.dependencies[payload.rootDependencyName].versions);
}

const payloadRootDependencyVersions = payloadRootVersions(payload);
const comparePayloadRootDependencyVersions = payloadRootVersions(comparePayload);
if (util.isDeepStrictEqual(payloadRootDependencyVersions, comparePayloadRootDependencyVersions) {
  // (should be) same payloads
}
```

Reply:
I am not sure, @fraxken any idea?

Reply:
Could be dangerous for CWD, because we're not thinking of upgrading the project version, but the deps are evolving.

Also because of SemVer resolution I think even if the package has the same version something could changes. So safer to keep it like that.